frontend/src/components/Authentication/Signup.js

```diff
@@ -20,6 +20,44 @@ const Signup = () => {
   const [password, setPassword] = useState();
   const [pic, setPic] = useState();
   const [picLoading, setPicLoading] = useState(false);
+  const [passwordStrength, setPasswordStrength] = useState("");
+
+  const checkPasswordStrength = (password)=>{
+    const hasUpperCase = /[A-Z]/.test(password)
+    const hasLowerCase = /[a-z]/.test(password)
+    const hasNumbers = /[0-9]/.test(password)
+    const hasSymbols = /[!@#$%^&*()_+\-=\[\]{};':"\\|,.<>\/?]+/.test(password)
+
+    if(password.length<6){
+      return "weak";
+    }
+    else if(password.length<8){
+      return "medium";
+    }
+    else if(hasUpperCase && hasLowerCase && hasNumbers && hasSymbols){
+      return "strong";
+    }
+    else{
+      return "medium";
+    }
+  }
+  const handlePasswordChange = (e)=>{
+    const newPassword = e.target.value;
+    setPassword(newPassword);
+    setPasswordStrength(checkPasswordStrength(newPassword));
+  }
+  const getPasswordStrengthColor=()=>{
```

Comment:
U have added color make it like the user can't signup without giving strong password and give a hidden response when the user enters low strength password
like 
![image](https://github.com/user-attachments/assets/15a9725f-5f38-47ff-b812-7b1cecfe2a06)

like here